_rules/css-orientation-lock-media-queries-b33eff.md

```diff
@@ -0,0 +1,199 @@
+---
+id: b33eff
+name: Page has no orientation lock specified using CSS media queries
+rule_type: atomic
+description: |
+  This rule checks that page content is not locked to any single display orientation using CSS media queries.
+accessibility_requirements:
+ wcag20:1.3.4: # Orientation
+  forConformance: true
+  failed: not satisfied
+  passed: further testing needed
+  inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any element that is [visible](#visible) and has a CSS `transform` property with a [rotate](https://drafts.csswg.org/css-transforms/#funcdef-transform-rotate) transform-function, set from a CSS media query with an [orientation](https://drafts.csswg.org/mediaqueries-3/#orientation) media feature (`landscape` or `portrait`).
```

Comment:
This does not account for 3D transformations: https://drafts.csswg.org/css-transforms-2/#transform-functions

Reply:
@kasperisager the applicability specifies [rotate](https://drafts.csswg.org/css-transforms/#funcdef-transform-rotate) which only refer to **2D Transform Functions**.  So this rule does not cater to 3D transforms.

Reply:
That's exactly my point; it doesn't, and I absolutely think that it should.

Reply:
Looking at this again, the rule really needs to look at the resulting transformation matrix (https://drafts.csswg.org/css-transforms/#transformation-matrix) instead of the value of the `transform` property. There are several reasons for this:

1. A transformation matrix can be passed explicitly using the `matrix()` function.
2. A 2D rotation (`rotate()`) is really just a 3D rotation (`rotate3d()`) with a fixed direction (`[0, 0, 1]`).
3. The value of the `transform` property isn't necessarily valid. Consider this:
  ```css
  { transform: rotate(90deg) doh; }
  ```